qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -104,24 +104,51 @@ class OptimizationResult:
         results: The original results object returned from the optimization algorithm. This can
             contain more information than only the optimal value and function value.
         status: The termination status of the algorithm.
+        variables: The list of variables under optimization.
     """
 
     Status = OptimizationResultStatus
 
     def __init__(self, x: Optional[Any] = None, fval: Optional[Any] = None,
                  results: Optional[Any] = None,
-                 status: OptimizationResultStatus = OptimizationResultStatus.SUCCESS) -> None:
-        self._val = x
-        self._fval = fval
-        self._results = results
-        self._status = status
+                 status: OptimizationResultStatus = OptimizationResultStatus.SUCCESS,
+                 variables: Optional[List[Variable]] = None) -> None:
```

Comment:
I don't think `variables` may be `None`. It makes APIs behave tricky in such a case.